controllers/cache.go

```diff
@@ -0,0 +1,27 @@
+package controllers
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CacheCreator(conf *rest.Config, opts cache.Options) (cache.Cache, error) {
+	labelSelector, err := labels.Parse("app.kubernetes.io/managed-by in (btp-manager,kcp-kyma-environment-broker)")
+	if err != nil {
+		panic(fmt.Sprintf("unable to parse label selector: %s", err))
+	}
+	objSelector := cache.ObjectSelector{
+		Label: labelSelector,
+	}
+
+	opts.SelectorsByObject = map[client.Object]cache.ObjectSelector{
+		&corev1.Secret{}:    objSelector,
```

Comment:
Have you limited that to secrets and configmaps only by purpose? I mean we have webhooks too.